src/py42/modules/securitydata.py

```diff
@@ -169,6 +178,82 @@ def search_file_events(self, query):
         file_event_client = self._microservices_client_factory.get_file_event_client()
         return file_event_client.search(query)
 
+    def _search_by_hash(self, hash, type):
+        query = FileEventQuery.all(type.eq(hash))
+        response = self.search_file_events(query)
+        return response[u"fileEvents"]
+
+    def _find_file_versions(self, md5_hash, sha256_hash):
+        file_event_client = self._microservices_client_factory.get_file_event_client()
+        pds_client = self._microservices_client_factory.get_preservation_data_service_client()
+        response = file_event_client.get_file_location_detail_by_sha256(sha256_hash)
+
+        if u"locations" not in response and not len(response[u"locations"]):
+            raise Py42ArchiveFileNotFoundError(u"PDS service can't find requested file.")
+
+        for device_id, paths in _parse_file_location_response(response):
+            try:
+                yield pds_client.find_file_versions(md5_hash, sha256_hash, device_id, paths)
+            except Py42HTTPError as err:
+                # API searches multiple paths to find the file to be streamed, as returned by
+                # 'get_file_location_detail_by_sha256', hence we keep looking until we find a stream
+                # to return
+                debug.logger.warning(u"PDS fetch file version API failed. Error: ", err)
```

Comment:
this log message should contain the hash somewhere.

Reply:
Added. Do note, had to add both the hashes in this particular case, otherwise I had to pass the hash-type to indicate which hash method is requested. I was aware that passing requested hash values is important  but left it out since I felt its an overkill to pass variable only to set an error message. 

I am open to rework again, If needed I can do the change accordingly.